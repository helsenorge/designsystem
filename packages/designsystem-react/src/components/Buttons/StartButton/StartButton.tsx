import React, {useState} from 'react';
import {HTMLButtonProps, HTMLAnchorProps} from '../../../constants';
import Button, {ButtonVariants, ButtonSizes} from '../Button/Button';

type StartButtonVariants = 'primary' | 'secondary' | 'tertiary';
type StartButtonTag = 'a' | 'button';

interface StartButtonProps extends HTMLButtonProps, HTMLAnchorProps {
  variant?: StartButtonVariants;
  children: React.ReactNode;
  htmlTag?: StartButtonTag;
  loading?: boolean;
  onClick?: () => void;
}

const variantMap = {
  primary: 'fill',
  secondary: 'fill',
  tertiary: 'outline',
};

const sizeMap = {
  primary: 'large',
  secondary: 'medium',
  tertiary: 'medium',
};

const StartButton = React.forwardRef((props: StartButtonProps, ref: any) => {
  const {children, variant = 'primary', htmlTag = 'button', loading, ...rest} = props;
  return (
    <Button
      variant={variantMap[variant] as ButtonVariants}
      size={sizeMap[variant] as ButtonSizes}
      htmlTag={htmlTag}
      loading={loading}
      color="vein"
      {...rest}>
      {children}
    </Button>
  );
});

export default StartButton;
