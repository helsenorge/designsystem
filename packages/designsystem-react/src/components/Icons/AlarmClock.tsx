import React from 'react';
import {IconRawProps} from './Icon';

const AlarmClock = React.forwardRef((svgProps: IconRawProps, ref: any) => {
  const {size = 48, color = 'black', hoverColor = 'black', isHovered = false, ...props} = svgProps;
  return !isHovered ? (
    <svg width={size} height={size} viewBox="0 0 38 38" ref={ref} className="icon" {...props}>
      <path
        fill={color}
        fillRule="evenodd"
        d="M18.8962,26.912 C14.6852,26.912 11.2602,23.487 11.2602,19.277 C11.2602,18.004 11.5782,16.805 12.1312,15.748 C12.3322,15.364 12.5672,15.003 12.8272,14.662 C12.8512,14.63 12.8742,14.597 12.8992,14.566 C13.1562,14.239 13.4402,13.936 13.7462,13.656 C13.7802,13.625 13.8142,13.593 13.8492,13.563 C14.1632,13.285 14.4992,13.033 14.8562,12.81 C14.8832,12.793 14.9102,12.777 14.9372,12.761 C15.3112,12.533 15.7052,12.334 16.1172,12.173 C16.9802,11.834 17.9152,11.642 18.8962,11.642 C19.8792,11.642 20.8162,11.835 21.6802,12.175 C22.0892,12.336 22.4802,12.533 22.8512,12.759 C22.8802,12.776 22.9082,12.793 22.9372,12.811 C23.2932,13.034 23.6292,13.286 23.9422,13.563 C23.9772,13.593 24.0102,13.624 24.0432,13.655 C24.3492,13.935 24.6342,14.238 24.8912,14.565 C24.9172,14.597 24.9402,14.63 24.9642,14.662 C25.2232,15.002 25.4582,15.362 25.6582,15.745 C26.2132,16.802 26.5312,18.002 26.5312,19.277 C26.5312,23.487 23.1062,26.912 18.8962,26.912 L18.8962,26.912 Z M10.2481971,12.476 C10.2472,11.797 10.5092,11.158 10.9892,10.677 C11.4682,10.196 12.1072,9.931 12.7862,9.93 L12.7912,9.93 C13.6912,9.93 14.4992,10.404 14.9492,11.156 C14.9272,11.166 14.9092,11.181 14.8882,11.191 C14.1862,11.541 13.5362,11.977 12.9522,12.489 C12.8892,12.544 12.8292,12.6 12.7682,12.656 C12.5142,12.891 12.2712,13.139 12.0452,13.401 C12.0222,13.429 11.9962,13.453 11.9732,13.481 C11.7262,13.775 11.5012,14.089 11.2932,14.414 C11.2782,14.437 11.2592,14.457 11.2442,14.481 C10.6252,14.007 10.2492,13.271 10.2481971,12.476 L10.2481971,12.476 Z M24.9982,9.93 L25.0022,9.93 C25.6792,9.93 26.3162,10.193 26.7962,10.671 C27.2772,11.151 27.5422,11.789 27.543202,12.468 C27.5442,13.266 27.1672,14.006 26.5472,14.48 C26.5322,14.456 26.5122,14.435 26.4972,14.411 C26.2882,14.086 26.0632,13.773 25.8162,13.478 C25.7952,13.454 25.7722,13.431 25.7512,13.407 C25.5222,13.14 25.2762,12.89 25.0192,12.652 C24.9602,12.598 24.9022,12.544 24.8422,12.491 C24.2582,11.979 23.6092,11.543 22.9072,11.193 C22.8852,11.182 22.8652,11.167 22.8432,11.156 C23.2922,10.405 24.0982,9.931 24.9982,9.93 L24.9982,9.93 Z M27.9312,19.277 C27.9312,18.016 27.6702,16.815 27.2012,15.724 C28.2762,14.999 28.9462,13.788 28.9442045,12.466 C28.9422,11.412 28.5302,10.423 27.7852,9.68 C27.0412,8.938 26.0522,8.53 25.0022,8.53 L24.9962,8.53 C23.5172,8.532 22.2052,9.357 21.5362,10.637 C20.7002,10.381 19.8142,10.242 18.8962,10.242 C17.9782,10.242 17.0922,10.381 16.2562,10.637 C15.5862,9.354 14.2712,8.53 12.7912,8.53 L12.7842,8.53 C11.7302,8.532 10.7412,8.943 9.9982,9.689 C9.2542,10.434 8.8452,11.425 8.8471927,12.478 C8.8502,13.795 9.5182,15.001 10.5902,15.725 C10.1212,16.816 9.8602,18.016 9.8602,19.277 C9.8602,22.195 11.2562,24.787 13.4102,26.44 L11.8862,28.087 L12.9142,29.037 L14.5942,27.221 C15.8742,27.917 17.3392,28.312 18.8962,28.312 C20.4522,28.312 21.9182,27.917 23.1982,27.221 L24.8772,29.037 L25.9052,28.087 L24.3812,26.44 C26.5342,24.787 27.9312,22.195 27.9312,19.277 L27.9312,19.277 Z M18.1962,19.215 L15.5472,19.215 L15.5472,20.615 L19.5962,20.615 L19.5962,15.552 L18.1962,15.552 L18.1962,19.215 Z"
      />
    </svg>
  ) : (
    <svg width={size} height={size} viewBox="0 0 38 38" ref={ref} className="icon" {...props}>
      <path
        fill={hoverColor}
        fillRule="evenodd"
        d="M26.5478,14.4809 C26.5308,14.4549 26.5098,14.4319 26.4928,14.4059 C26.2868,14.0859 26.0658,13.7769 25.8218,13.4859 C25.7928,13.4519 25.7608,13.4209 25.7318,13.3869 C25.5118,13.1319 25.2768,12.8919 25.0298,12.6619 C24.9648,12.6019 24.9008,12.5429 24.8338,12.4849 C24.2518,11.9749 23.6058,11.5409 22.9068,11.1929 C22.8848,11.1819 22.8648,11.1669 22.8428,11.1559 C23.2928,10.4039 24.1008,9.9299 25.0008,9.9299 L25.0058,9.9299 C25.6848,9.9309 26.3228,10.1959 26.8028,10.6769 C27.2818,11.1579 27.5448,11.7969 27.5438028,12.4759 C27.5428,13.2719 27.1658,14.0069 26.5478,14.4809 L26.5478,14.4809 Z M18.8958,26.9119 C14.6858,26.9119 11.2608,23.4869 11.2608,19.2769 C11.2608,18.0039 11.5778,16.8049 12.1308,15.7489 C12.3318,15.3639 12.5678,15.0029 12.8278,14.6619 C12.8518,14.6299 12.8748,14.5979 12.8998,14.5659 C13.1568,14.2389 13.4418,13.9349 13.7478,13.6549 C13.7808,13.6249 13.8138,13.5939 13.8488,13.5629 C14.1628,13.2859 14.4998,13.0319 14.8578,12.8089 C14.8838,12.7929 14.9098,12.7769 14.9358,12.7619 C15.3108,12.5339 15.7048,12.3339 16.1188,12.1719 C16.9808,11.8339 17.9148,11.6419 18.8958,11.6419 C19.8768,11.6419 20.8128,11.8349 21.6758,12.1739 C22.0868,12.3349 22.4778,12.5329 22.8508,12.7589 C22.8798,12.7769 22.9098,12.7939 22.9388,12.8119 C23.2928,13.0339 23.6258,13.2849 23.9378,13.5589 C23.9758,13.5929 24.0128,13.6269 24.0498,13.6609 C24.3518,13.9379 24.6328,14.2379 24.8878,14.5609 C24.9158,14.5969 24.9428,14.6349 24.9708,14.6709 C25.2258,15.0079 25.4578,15.3639 25.6568,15.7429 C26.2118,16.7999 26.5298,18.0009 26.5298,19.2769 C26.5298,23.4869 23.1058,26.9119 18.8958,26.9119 L18.8958,26.9119 Z M10.2477941,12.4759 C10.2448,11.0739 11.3838,9.9319 12.7848,9.9299 L12.7898,9.9299 C13.6908,9.9299 14.4988,10.4039 14.9488,11.1559 C14.9288,11.1659 14.9108,11.1799 14.8898,11.1899 C14.1868,11.5399 13.5358,11.9769 12.9508,12.4899 C12.8898,12.5439 12.8298,12.5989 12.7698,12.6549 C12.5148,12.8909 12.2708,13.1389 12.0438,13.4039 C12.0218,13.4299 11.9968,13.4539 11.9748,13.4799 C11.7268,13.7749 11.5018,14.0889 11.2928,14.4139 C11.2778,14.4369 11.2588,14.4579 11.2438,14.4809 C10.6258,14.0079 10.2488,13.2719 10.2477941,12.4759 L10.2477941,12.4759 Z M28.9438073,12.4779 C28.9458,11.4249 28.5368,10.4349 27.7938,9.6889 C27.0508,8.9429 26.0618,8.5319 25.0078,8.5299 L25.0008,8.5299 C23.5198,8.5299 22.2058,9.3539 21.5348,10.6369 C20.6998,10.3809 19.8138,10.2419 18.8958,10.2419 C17.9778,10.2419 17.0918,10.3809 16.2568,10.6369 C15.5858,9.3539 14.2718,8.5299 12.7898,8.5299 L12.7828,8.5299 C10.6098,8.5339 8.8448,10.3049 8.84779618,12.4779 C8.8498,13.7959 9.5188,15.0019 10.5898,15.7249 C10.1218,16.8159 9.8608,18.0159 9.8608,19.2769 C9.8608,22.1949 11.2568,24.7859 13.4098,26.4399 L11.8858,28.0859 L12.9128,29.0379 L14.5938,27.2209 C15.8738,27.9169 17.3398,28.3129 18.8958,28.3129 C20.4518,28.3129 21.9168,27.9169 23.1968,27.2209 L24.8778,29.0379 L25.9048,28.0859 L24.3808,26.4399 C26.5338,24.7859 27.9308,22.1949 27.9308,19.2769 C27.9308,18.0159 27.6698,16.8159 27.2008,15.7249 C28.2728,15.0019 28.9418,13.7949 28.9438073,12.4779 L28.9438073,12.4779 Z M31.0508,6.8599 C30.1168,5.7649 28.8188,5.1559 27.3928,5.1479 L27.3848,6.5479 C28.6818,6.5559 29.5138,7.2159 29.9848,7.7679 C30.6038,8.4959 30.9558,9.4549 30.9498,10.3989 L32.3508,10.4069 C32.3578,9.1319 31.8838,7.8399 31.0508,6.8599 L31.0508,6.8599 Z M10.3998,5.1479 C7.3188,5.1659 5.4268,7.8919 5.4408,10.4069 L6.8408,10.3989 C6.8308,8.5569 8.1918,6.5609 10.4068,6.5479 L10.3998,5.1479 Z M18.1948,19.2149 L15.5468,19.2149 L15.5468,20.6149 L19.5958,20.6149 L19.5958,15.5519 L18.1948,15.5519 L18.1948,19.2149 Z"
      />
    </svg>
  );
});

export default AlarmClock;
