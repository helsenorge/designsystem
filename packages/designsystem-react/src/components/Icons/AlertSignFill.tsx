import React from 'react';
import {IconRawProps} from './Icon';

const AlertSignFill = React.forwardRef((svgProps: IconRawProps, ref: any) => {
  const hasHover = false;
  const {size = 48, color = 'black', hoverColor = 'black', isHovered = false, ...props} = svgProps;
  return !isHovered || !hasHover ? (
    <svg width={size} height={size} viewBox="0 0 48 48" ref={ref} className="icon" {...props}>
      <path
        fill={color}
        fillRule="evenodd"
        d="M24.8975,21.447 L24.8975,28.675 C24.8975,29.168 24.4975,29.568 24.0045,29.568 C23.5115,29.568 23.1125,29.168 23.1125,28.675 L23.1125,21.447 C23.1125,20.954 23.5115,20.554 24.0045,20.554 C24.4975,20.554 24.8975,20.954 24.8975,21.447 L24.8975,21.447 Z M25.3985,33.048 C25.3985,33.817 24.7745,34.441 24.0045,34.441 C23.2355,34.441 22.6115,33.817 22.6115,33.048 C22.6115,32.278 23.2355,31.654 24.0045,31.654 C24.7745,31.654 25.3985,32.278 25.3985,33.048 L25.3985,33.048 Z M6.5765,39.758 L41.4235,39.758 L24.1115,8.242 L6.5765,39.758 Z"
      />
    </svg>
  ) : (
    <svg width={size} height={size} viewBox="0 0 48 48" ref={ref} className="icon" {...props}></svg>
  );
});

export default AlertSignFill;
