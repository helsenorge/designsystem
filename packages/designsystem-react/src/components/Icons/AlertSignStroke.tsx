import React from 'react';
import {IconRawProps} from './Icon';

const AlertSignStroke = React.forwardRef((svgProps: IconRawProps, ref: any) => {
  const {size = 48, color = 'black', hoverColor = 'black', isHovered = false, ...props} = svgProps;
  const isExtraSmall = size <= 38;
  const normal = (
    <path
      fillRule="evenodd"
      d="M24.0035,31.578 C23.2335,31.578 22.6095,32.202 22.6095,32.972 C22.6095,33.741 23.2335,34.365 24.0035,34.365 C24.7735,34.365 25.3975,33.741 25.3975,32.972 C25.3975,32.202 24.7735,31.578 24.0035,31.578 L24.0035,31.578 Z M24.1005,11.388 L38.7875,38.123 L9.2265,38.123 L24.1005,11.388 Z M24.1085,8.459 L6.8165,39.541 L41.1835,39.541 L24.1085,8.459 Z M24.0035,29.492 C24.4965,29.492 24.8965,29.092 24.8965,28.599 L24.8965,21.371 C24.8965,20.878 24.4965,20.478 24.0035,20.478 C23.5105,20.478 23.1105,20.878 23.1105,21.371 L23.1105,28.599 C23.1105,29.092 23.5105,29.492 24.0035,29.492 L24.0035,29.492 Z"
    />
  );

  const normalHover = (
    <path
      fillRule="evenodd"
      d="M24.0035,31.578 C23.2335,31.578 22.6095,32.202 22.6095,32.972 C22.6095,33.741 23.2335,34.365 24.0035,34.365 C24.7735,34.365 25.3975,33.741 25.3975,32.972 C25.3975,32.202 24.7735,31.578 24.0035,31.578 L24.0035,31.578 Z M24.1005,11.388 L38.7875,38.123 L9.2265,38.123 L24.1005,11.388 Z M24.1085,8.459 L6.8165,39.541 L41.1835,39.541 L24.1085,8.459 Z M24.0035,27.492 C24.4965,27.492 24.8965,27.092 24.8965,26.599 L24.8965,19.371 C24.8965,18.878 24.4965,18.478 24.0035,18.478 C23.5105,18.478 23.1105,18.878 23.1105,19.371 L23.1105,26.599 C23.1105,27.092 23.5105,27.492 24.0035,27.492 L24.0035,27.492 Z"
    />
  );

  const simplified = (
    <path
      fillRule="evenodd"
      d="M24.0034926,31.5620463 C23.2351137,31.5620463 22.6124274,32.1847326 22.6124274,32.9531116 C22.6124274,33.7204926 23.2351137,34.3431789 24.0034926,34.3431789 C24.7718716,34.3431789 25.3945579,33.7204926 25.3945579,32.9531116 C25.3945579,32.1847326 24.7718716,31.5620463 24.0034926,31.5620463 L24.0034926,31.5620463 Z M24.1002884,11.4145516 L38.7563684,38.0932674 L9.25760211,38.0932674 L24.1002884,11.4145516 Z M24.1082716,8.49171789 L6.85267579,39.5082821 L41.1473242,39.5082821 L24.1082716,8.49171789 Z M24.0034926,29.4804379 C24.4954547,29.4804379 24.8946126,29.08128 24.8946126,28.5893179 L24.8946126,21.3765347 C24.8946126,20.8845726 24.4954547,20.4854147 24.0034926,20.4854147 C23.5115305,20.4854147 23.1123726,20.8845726 23.1123726,21.3765347 L23.1123726,28.5893179 C23.1123726,29.08128 23.5115305,29.4804379 24.0034926,29.4804379 L24.0034926,29.4804379 Z"
    />
  );

  const simplifiedHover = (
    <path
      fillRule="evenodd"
      d="M24.0034926,31.5620463 C23.2351137,31.5620463 22.6124274,32.1847326 22.6124274,32.9531116 C22.6124274,33.7204926 23.2351137,34.3431789 24.0034926,34.3431789 C24.7718716,34.3431789 25.3945579,33.7204926 25.3945579,32.9531116 C25.3945579,32.1847326 24.7718716,31.5620463 24.0034926,31.5620463 L24.0034926,31.5620463 Z M24.1002884,11.4145516 L38.7563684,38.0932674 L9.25760211,38.0932674 L24.1002884,11.4145516 Z M24.1082716,8.49171789 L6.85267579,39.5082821 L41.1473242,39.5082821 L24.1082716,8.49171789 Z M24.0034926,26.9541221 C24.4954547,26.9541221 24.8946126,26.5549642 24.8946126,26.0630021 L24.8946126,18.8502189 C24.8946126,18.3582568 24.4954547,17.9590989 24.0034926,17.9590989 C23.5115305,17.9590989 23.1123726,18.3582568 23.1123726,18.8502189 L23.1123726,26.0630021 C23.1123726,26.5549642 23.5115305,26.9541221 24.0034926,26.9541221 L24.0034926,26.9541221 Z"
    />
  );

  return (
    <svg
      width={size}
      height={size}
      fill={isHovered ? hoverColor : color}
      viewBox="0 0 48 48"
      ref={ref}
      className="icon"
      {...props}>
      {isExtraSmall ? (isHovered ? simplifiedHover : simplified) : isHovered ? normalHover : normal}
    </svg>
  );
});

export default AlertSignStroke;
