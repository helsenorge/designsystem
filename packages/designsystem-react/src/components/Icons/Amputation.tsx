import React from 'react';
import {IconRawProps} from './Icon';

const Amputation = React.forwardRef((svgProps: IconRawProps, ref: any) => {
  const {size = 48, color = 'black', hoverColor = 'black', isHovered = false, ...props} = svgProps;
  const isExtraSmall = size <= 38;
  const normal = (
    <path
      fillRule="evenodd"
      d="M23.8777 28.8506C23.5257 28.9736 23.1487 28.9466 22.8167 28.7706 22.4857 28.5946 22.2517 28.2986 22.1507 27.9146L19.4237 18.8236C19.3317 18.4746 18.9927 18.2506 18.6367 18.2926 18.3117 18.3346 18.0087 18.6796 18.0097 19.0076L18.0427 26.9826 13.6667 28.4306 13.6557 27.1316 13.6527 27.0136C13.6447 26.6546 13.3537 26.3686 12.9867 26.3806 12.6337 26.3896 12.3537 26.6786 12.3537 27.0306 12.3537 27.0366 12.3537 27.0406 12.3547 27.0466L12.3647 39.3376C12.3607 40.5366 11.3817 41.5106 10.1837 41.5106 8.9817 41.5106 8.0037 40.5356 8.0017 39.3376L8.0597 19.0046C8.0597 18.6446 7.7897 18.3386 7.4327 18.2926 7.4087 18.2896 7.3857 18.2876 7.3617 18.2876 7.0637 18.2876 6.7347 18.5196 6.6497 18.8016L3.9117 27.9366C3.8167 28.2986 3.5827 28.5946 3.2517 28.7706 2.9207 28.9476 2.5437 28.9746 2.1907 28.8506 1.5457 28.6236 1.1807 27.9516 1.3387 27.2956L3.9747 17.0986C4.6067 14.4996 7.8067 12.7546 10.3997 12.7546L15.6677 12.7546C18.2617 12.7546 21.4617 14.4996 22.0967 17.1086L24.7267 27.2866C24.8887 27.9516 24.5237 28.6236 23.8777 28.8506L23.8777 28.8506zM15.8857 35.0156C14.9117 35.0156 14.0927 34.3676 13.8147 33.4826L18.0657 32.1766 18.0667 32.8386C18.0647 34.0386 17.0857 35.0156 15.8857 35.0156L15.8857 35.0156zM16.9587 31.4706L13.7007 32.4716 13.6817 30.1746 16.9587 31.4706zM18.0597 30.8296L14.2037 29.3056 18.0477 28.0346 18.0597 30.8296zM25.9867 26.9696L23.3567 16.7916C22.5557 13.4956 18.7417 11.4546 15.6677 11.4546L10.3997 11.4546C7.3257 11.4546 3.5127 13.4956 2.7137 16.7826L.0777 26.9796C-.2413 28.2966.4817 29.6286 1.7617 30.0776 2.4497 30.3196 3.2157 30.2606 3.8617 29.9176 4.5077 29.5756 4.9837 28.9736 5.1637 28.2886L6.7477 23.0036 6.7007 39.3366C6.7047 41.2526 8.2677 42.8116 10.1837 42.8116 12.0967 42.8116 13.6577 41.2546 13.6657 39.3396L13.6627 35.5106C14.2667 36.0136 15.0417 36.3166 15.8857 36.3166 17.8007 36.3166 19.3627 34.7576 19.3677 32.8376L19.3267 23.0236 20.8987 28.2666C21.0847 28.9736 21.5617 29.5756 22.2067 29.9176 22.8537 30.2606 23.6187 30.3196 24.3077 30.0776 25.5877 29.6286 26.3107 28.2966 25.9867 26.9696L25.9867 26.9696zM12.907 2.2646C14.59 2.2646 15.958 3.6326 15.958 5.3156 15.958 6.9976 14.59 8.3676 12.907 8.3676 11.224 8.3676 9.855 6.9976 9.855 5.3156 9.855 3.6326 11.224 2.2646 12.907 2.2646M12.907 9.6676C15.306 9.6676 17.259 7.7146 17.259 5.3156 17.259 2.9156 15.306.9636 12.907.9636 10.508.9636 8.554 2.9156 8.554 5.3156 8.554 7.7146 10.508 9.6676 12.907 9.6676"
      transform="translate(11 2)"
    />
  );

  const normalHover = (
    <path
      fillRule="evenodd"
      d="M21.7039 28.4277C21.3349 28.4947 20.9679 28.4067 20.6689 28.1807 20.3699 27.9547 20.1849 27.6257 20.1499 27.2687L19.4429 17.9397C19.4399 17.9027 19.4339 17.8657 19.4259 17.8307 19.3379 17.4797 19.0219 17.2407 18.6409 17.2917 18.3139 17.3317 18.0089 17.6787 18.0099 18.0077L18.0429 25.9837 13.6669 27.4307 13.6559 26.1317 13.6529 26.0137C13.6439 25.6547 13.3349 25.3677 12.9869 25.3807 12.6339 25.3897 12.3529 25.6787 12.3529 26.0307 12.3529 26.0357 12.3529 26.0407 12.3539 26.0467L12.3649 38.3377C12.3609 39.5357 11.3819 40.5107 10.1829 40.5107 8.9819 40.5107 8.0039 39.5347 8.0019 38.3377L8.0589 18.0047C8.0589 17.6447 7.7899 17.3387 7.4329 17.2927 7.4089 17.2897 7.3859 17.2877 7.3619 17.2877 7.0629 17.2877 6.7349 17.5197 6.6499 17.8017L3.9119 26.9367C3.8169 27.2987 3.5829 27.5947 3.2519 27.7707 2.9199 27.9477 2.5439 27.9747 2.1909 27.8507 1.5459 27.6237 1.1809 26.9507 1.3399 26.2957L3.9749 16.0987C4.6059 13.4997 7.8069 11.7547 10.3999 11.7547L15.6679 11.7547C18.2439 11.7547 21.4199 13.4767 22.0809 16.0457L22.7899 27.0177C22.8439 27.6997 22.3769 28.3057 21.7039 28.4277L21.7039 28.4277zM15.8859 34.0157C14.9119 34.0157 14.0929 33.3667 13.8139 32.4827L18.0649 31.1767 18.0669 31.8387C18.0639 33.0387 17.0859 34.0157 15.8859 34.0157L15.8859 34.0157zM16.9589 30.4707L13.7009 31.4717 13.6819 29.1747 16.9589 30.4707zM18.0589 29.8297L14.2039 28.3057 18.0479 27.0347 18.0589 29.8297zM23.3739 15.9037C23.3719 15.8657 23.3659 15.8277 23.3569 15.7917 22.5569 12.4947 18.7429 10.4537 15.6679 10.4537L10.3999 10.4537C7.3259 10.4537 3.5129 12.4947 2.7149 15.7817L.0789 25.9797C-.2411 27.2967.4809 28.6287 1.7619 29.0767 2.4499 29.3197 3.2159 29.2607 3.8619 28.9177 4.5079 28.5747 4.9839 27.9737 5.1639 27.2877L6.7479 22.0037 6.7009 38.3367C6.7049 40.2527 8.2669 41.8117 10.1829 41.8117 12.0969 41.8117 13.6579 40.2547 13.6659 38.3397L13.6629 34.5107C14.2669 35.0137 15.0419 35.3167 15.8859 35.3167 17.8009 35.3167 19.3629 33.7577 19.3679 31.8377L19.3539 28.6807C19.5039 28.8827 19.6819 29.0647 19.8859 29.2187 20.4679 29.6577 21.2169 29.8327 21.9339 29.7067 23.2689 29.4657 24.1929 28.2647 24.0859 26.9237L23.3739 15.9037zM14.7664 1.7129C16.4494 1.7129 17.8184 3.0809 17.8184 4.7639 17.8184 6.4459 16.4494 7.8159 14.7664 7.8159 13.0834 7.8159 11.7154 6.4459 11.7154 4.7639 11.7154 3.0809 13.0834 1.7129 14.7664 1.7129M14.7664 9.1159C17.1654 9.1159 19.1194 7.1629 19.1194 4.7639 19.1194 2.3639 17.1654.4119 14.7664.4119 12.3674.4119 10.4144 2.3639 10.4144 4.7639 10.4144 7.1629 12.3674 9.1159 14.7664 9.1159"
      transform="translate(11 3)"
    />
  );

  return (
    <svg
      width={size}
      height={size}
      fill={isHovered ? hoverColor : color}
      viewBox="0 0 48 48"
      ref={ref}
      className="icon"
      {...props}>
      {isHovered ? normalHover : normal}
    </svg>
  );
});

export default Amputation;
