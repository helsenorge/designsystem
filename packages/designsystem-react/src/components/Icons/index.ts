export {Icon, IconSize} from './Icon';
