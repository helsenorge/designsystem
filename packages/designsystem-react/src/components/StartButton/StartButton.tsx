import React from 'react';
import cx from 'classnames';
import {ButtonProps, Button} from '../Button';

interface StartButtonProps extends ButtonProps {}

// TODO: Move most of the logic out in a generic Button-component that the others inherit.
const StartButton = React.forwardRef((props: StartButtonProps, ref: any) => {
  const {children, variant, ...rest} = props;
  const classes = cx('is-start', {[`is-${variant}`]: variant === 'secondary' || variant === 'tertiary'});
  return (
    <Button className={classes} ref={ref} {...rest}>
      {children}
    </Button>
  );
});

export {StartButton};
