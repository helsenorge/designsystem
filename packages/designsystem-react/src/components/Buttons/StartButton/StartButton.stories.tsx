import React from 'react';
import {withKnobs, select, text} from '@storybook/addon-knobs';
import {storiesOf} from '@storybook/react';
import StartButton from './StartButton';
import {Icon} from '../../Icons';
import {allIcons} from '../../../../.storybook/knobs';

const stories = storiesOf('Buttons/StartButton', module);
stories.addDecorator(withKnobs);

stories.add('Default', () => (
  <StartButton variant={select('Variant', ['primary', 'secondary', 'tertiary'], 'primary')}>
    {text('Text', 'StartButton')}
  </StartButton>
));
