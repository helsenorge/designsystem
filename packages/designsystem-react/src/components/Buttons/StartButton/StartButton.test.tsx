import React from 'react';
import {render} from '@testing-library/react';
import StartButton from './StartButton';
import {Icon} from '../../Icons';

test('renders default variant correctly', (): void => {
  const {container} = render(<StartButton>StartButton</StartButton>);
  expect(container).toMatchSnapshot();
});
