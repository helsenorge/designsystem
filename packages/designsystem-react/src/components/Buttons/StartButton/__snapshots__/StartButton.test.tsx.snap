// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders default variant correctly 1`] = `
<div>
  <button
    class="sc-AxiKw fynHZr"
    color="vein"
  >
    <span
      class="sc-AxjAm iMxqKn"
    >
      StartButton
    </span>
  </button>
</div>
`;
